File: moedict-android/app/src/main/java/com/yichiuan/moedict/AppComponent.java
package com.yichiuan.moedict;

import com.yichiuan.moedict.data.DataModule;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;


@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        AppModule.class,
        DataModule.class,
        BuilderModule.class})
public interface AppComponent {
    void inject(MoeApplication application);
}

File: moedict-android/app/src/main/java/com/yichiuan/moedict/AppComponent.kt
package com.yichiuan.moedict

import com.yichiuan.moedict.data.DataModule
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton


@Singleton
@Component(modules = arrayOf(AndroidSupportInjectionModule::class, AppModule::class,
        DataModule::class, BuilderModule::class))
interface AppComponent {
    fun inject(application: MoeApplication)
}
